perf(language_server): avoid cloning on message conversion

Sysix opened the pull request:
Tested it with one `no-debugger` rule on a file (100 times).

Main:
```
heaptrack stats:
        allocations:            172985
        leaked allocations:     580
        temporary allocations:  53112
```

This PR:

```
heaptrack stats:
        allocations:            170985
        leaked allocations:     580
        temporary allocations:  53111
```

camc314 commented:
### Merge activity

* **Sep 24, 8:57 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Sysix committed:
perf(language_server): avoid cloning on message conversion (#14058)

Tested it with one `no-debugger` rule on a file (100 times).

Main:
```
heaptrack stats:
        allocations:            172985
        leaked allocations:     580
        temporary allocations:  53112
```

This PR:

```
heaptrack stats:
        allocations:            170985
        leaked allocations:     580
        temporary allocations:  53111
```
